symfony/plugins/orangehrmTimePlugin/Dto/EmployeeReportsSearchFilterParams.php

```diff
@@ -0,0 +1,138 @@
+<?php
+/**
+ * OrangeHRM is a comprehensive Human Resource Management (HRM) System that captures
+ * all the essential functionalities required for any enterprise.
+ * Copyright (C) 2006 OrangeHRM Inc., http://www.orangehrm.com
+ *
+ * OrangeHRM is free software; you can redistribute it and/or modify it under the terms of
+ * the GNU General Public License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Time\Dto;
+
+use InvalidArgumentException;
+use OrangeHRM\Leave\Dto\DateRangeSearchFilterParams;
+
+class EmployeeReportsSearchFilterParams extends DateRangeSearchFilterParams
+{
+    public const ALLOWED_SORT_FIELDS = ['customer.name'];
+
+    public const INCLUDE_TIMESHEETS_APPROVED_ONLY = 'onlyApproved';
+    public const INCLUDE_TIMESHEETS_ALL = 'all';
+
+    public const INCLUDE_TIMESHEETS = [
+        self::INCLUDE_TIMESHEETS_APPROVED_ONLY,
+        self::INCLUDE_TIMESHEETS_ALL,
+    ];
+
+    public const TIMESHEET_STATE = [
+        'SUBMITTED',
+        'APPROVED',
+        'REJECTED'
+    ];
```

Comment:
As per @RajithaKumaras KT we can optimize this https://github.com/orangehrm/orangehrm/pull/1086/files#diff-bb31e6dc1dabbc89f27534ba496bfafc4bb4d36e76597add44ce25ddc6d43e0eR41

Reply:
done